room.go

```diff
@@ -266,9 +288,31 @@ func (r *Room) JoinWithToken(url, token string, opts ...ConnectOption) error {
 		opt(params)
 	}
 
-	joinRes, err := r.engine.Join(url, token, params)
-	if err != nil {
-		return err
+	var joinRes *livekit.JoinResponse
+	cloudHostname, _ := parseCloudURL(url)
+	if params.DisableRegionDiscovery || cloudHostname == "" {
+		var err error
+		joinRes, err = r.engine.Join(url, token, params)
+		if err != nil {
+			return err
+		}
+	} else {
+		for joinRes == nil {
+			bestURL, err := r.regionURLProvider.BestURL(cloudHostname, token)
+			if err != nil {
+				return err
```

Comment:
if for whatever reason we can't find the best url here, should this just log an error instead and let it continue which will just try to join using the latency record?

Reply:
I'm not sure about this one. In what scenario would the region settings endpoint not be available, but the websocket/signal connection would still go through? This would just add another step before eventual failure.

Reply:
changed the logic around to always fallback to the original url on any failure